hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/SignedStateHolder.java

```diff
@@ -115,7 +116,7 @@ public enum Validity {
      * @param validity - whether the contract is valid or note, aka active or deleted
      */
     public record Contract(
-            @NonNull Set</*@NonNull*/ Integer> ids, @NonNull byte[] bytecode, @NonNull Validity validity) {
+            @NonNull TreeSet</*@NonNull*/ Integer> ids, @NonNull byte[] bytecode, @NonNull Validity validity) {
```

Comment:
Is the `TreeSet` necessary? Would prefer to keep a `Set` if possible

Reply:
It _was_ a `Set` then I changed it - there's supposed to be a comment somewhere that says the _reason_ for this is to have a canonical ordering of ids (sorted) so that when _writing_ the file it is written in a deterministic order.  If the sorting wasn't done here it would be done elsewhere - and! It would be done in multiple places, because _another_ operation is _picking_ the "canonical" id that will represent a particular contract (so you can index them and so on ) and that (I chose to be) is the _lowest_ contract id of the set of contract ids.  To find it you'd have to sort or get the minimum anyway.

I'll find where I put the comment and put it here because it's obviously in the wrong place.

Reply:
Ah got it, that makes total sense, thanks!

Reply:
done